src/main/java/io/github/syst3ms/skriptparser/expressions/ExprHashedString.java

```diff
@@ -0,0 +1,79 @@
+package io.github.syst3ms.skriptparser.expressions;
+
+import io.github.syst3ms.skriptparser.Parser;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+import org.jetbrains.annotations.Nullable;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+
+/**
+ * Hashes the given text using the MD5 or SHA-256 algorithms.
+ * Each algorithm is suitable for different use cases.
+ * <ul>
+ * 	<li>MD5 is provided mostly for backwards compatibility, as it is outdated and not secure.</li>
```

Comment:
I'd rather not even allow MD5 usage. I doubt the amount of Skript use cases requiring MD5 backwards-compatibility is very high, and new implementations should be encouraged to use SHA-256. If someone wants to use MD5, they'll have to make an addon, let's not implement that security risk ourselves.